molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/utils/config/SecurityITConfig.java

```diff
@@ -0,0 +1,63 @@
+package org.molgenis.integrationtest.utils.config;
+
+import org.molgenis.auth.*;
+import org.molgenis.integrationtest.data.aggregation.AggregationTestConfig;
+import org.molgenis.security.core.utils.SecurityUtils;
+import org.molgenis.security.owned.OwnedEntityType;
+import org.molgenis.security.permission.*;
+import org.molgenis.security.settings.AuthenticationSettingsImpl;
+import org.molgenis.security.token.DataServiceTokenService;
+import org.molgenis.security.token.TokenGenerator;
+import org.molgenis.security.user.UserAccountServiceImpl;
+import org.molgenis.security.user.UserServiceImpl;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyAuthoritiesMapper;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.Collection;
+
+import static java.util.Collections.singleton;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@Configuration
+@EnableWebSecurity
+@Import({ PermissionServiceImpl.class, DataServiceTokenService.class, TokenGenerator.class, TokenFactory.class,
+		TokenMetaData.class, SecurityPackage.class, UserMetaData.class, OwnedEntityType.class,
+		UserAccountServiceImpl.class, UserServiceImpl.class, BCryptPasswordEncoder.class,
+		PermissionSystemServiceImpl.class, UserAuthorityFactory.class, UserAuthorityMetaData.class, UserFactory.class,
+		AuthorityMetaData.class, AggregationTestConfig.class, RoleHierarchyAuthoritiesMapper.class,
+		GroupAuthorityFactory.class, GroupAuthorityMetaData.class, GroupMetaData.class,
+		AuthenticationSettingsImpl.class, PrincipalSecurityContextRegistryImpl.class, SecurityContextRegistryImpl.class,
+		AuthenticationAuthoritiesUpdaterImpl.class })
+public class SecurityITConfig
+{
+	public final static String ANONYMOUSE_USER = "anonymouse";
+	public final static String SUPERUSER_NAME = "admin";
+	public final static String SUPERUSER_ROLE = "SU";
+	public final static String TOKEN_DESCRIPTION = "REST token";
+
+	@Bean
+	public RoleHierarchy roleHierarchy()
+	{
+		return mock(RoleHierarchy.class);
```

Comment:
Create technical debt GitHub issue to use role hierarchy instead of defining the whole hierarchy in your MockWithUser

Reply:
Issue #6843